### PR TITLE
[okta] Fix concurrent modification exception

### LINKED ISSUE #6506: [Okta] Ingest Pipeline Failing on Some Documents
While reviewing ingest pipeline performance metrics, I noticed that ~25% of our Okta documents are failing to run the ingest pipeline fully. It seems that when the pipeline fails, this isn't logged anywhere.

image

```
{
  "root_cause": [
    {
      "type": "concurrent_modification_exception",
      "reason": "concurrent_modification_exception: null"
    }
  ],
  "type": "script_exception",
  "reason": "runtime error",
  "script_stack": [
    "java.base/java.util.HashMap$HashIterator.nextNode(HashMap.java:1605)",
    "java.base/java.util.HashMap$EntryIterator.next(HashMap.java:1638)",
    "java.base/java.util.HashMap$EntryIterator.next(HashMap.java:1636)",
    "for (def e: de.entrySet()) {\n        ",
    "              ^---- HERE"
  ],
  "script": "def arr = ctx.okta?.target; ...",
  "lang": "painless",
  "position": {
    "offset": 342,
    "start": 328,
    "end": 365
  },
  "caused_by": {
    "type": "concurrent_modification_exception",
    "reason": "concurrent_modification_exception: null"
  }
}
```

the `error.message` field on the documents is empty instead of containing an error indicating that there is a problem:
image

Sample event where this occurs: 
```

```

### PR DESCRIPTION
**Pull request by andrewkroh**
## What does this PR do?

Use the entrySet().removeIf() lambda to perform the remove operation on a map instead of calling remove() on the map while looping over the entrySet() items.

Fixes #6506

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/okta/data_stream/system/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -564,11 +565,9 @@ processors:
...
             def de = arr[i].get("detailEntry");
             if (de != null) {
-              for (def e: de.entrySet()) {
-                if (!["methodTypeUsed","methodUsedVerifiedProperties"].contains(e.getKey())) {
-                  de.remove(e.getKey());
-                }
-              }
+              de.entrySet().removeIf(entry -> 
+                entry.getKey() != "methodTypeUsed" && 
+                entry.getKey() != "methodUsedVerifiedProperties");
```
Comment (efd6) [inline packages/okta/data_stream/system/elasticsearch/ingest_pipeline/default.yml:570]: Go has spoiled me with map [iteration](https://go.dev/ref/spec#For_statements) (3.) and [elem deletion](https://go.dev/ref/spec#Deletion_of_map_elements).

File: packages/okta/data_stream/system/_dev/test/pipeline/test-okta-system-events.log [hunk 2]
```diff
@@ -21,3 +21,4 @@
...
...ntext":{"debugData":{"requestId":"XXXXXXXXXXXXXXXXXX","requestUri":"/idp/authenticators","url":"/idp/authenticators?"}},"device":null,"displayMessage":"Add device to user","eventType":"device.user.add","legacyEventType":null,"outcome":{"reason":null,"result":"SUCCESS"},"published":"2023-06-07T15:49:45.109Z","request":{"ipChain":[{"geographicalContext":{"city":"Bay Shore","country":"United States","geolocation":{"lat":40.7051,"lon":-73.243},"postalCode":"11706","state":"New York"},"ip":"175.16.199.18","source":null,"version":"V4"}]},"securityContext":{"asNumber":701,"asOrg":"verizon","domain":"verizon.net","isProxy":false,"isp":"verizon"},"severity":"INFO","target":[{"alternateId":"unknown","detailEntry":{"deviceStatus":"CREATED","managed":"false","oktaDeviceId":"xxxxxxxxxxxxxxxxx","osPlatform":"MACOS","osVersion":"13.4.0","serialNumber":"XXXXXXXX","tpmPresent":"false","uuid":"AAAAAAAA-CCCC-DDDD-EEEE-BBBBBBBBBBBB"},"displayName":"John's MacBook Pro","id":"fakefakefakefake","type":"UDDevice"}],"transaction":{"detail":{"requestApiTokenId":"aa.aa.bbbbbbbbbbbbbbbbbbbbbbb_wwwwwwwwwwwwwwww"},"id":"ABCDEFCGALKDJDLK","type":"WEB"},"uuid":"aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa","version":"0"}
```
Comment (efd6) [inline packages/okta/data_stream/system/_dev/test/pipeline/test-okta-system-events.log:24]: Final newline.

**Review by efd6 — APPROVED** [review]
It's disturbing that this doesn't generate a {meaningful,any} error during execution.

File: packages/okta/data_stream/system/_dev/test/pipeline/test-okta-system-events.log (hunk 2 above)
Comment (andrewkroh) [reply packages/okta/data_stream/system/_dev/test/pipeline/test-okta-system-events.log]: Added.

**Pull request merged**